pkg/apiserver/resthandler.go

```diff
@@ -234,7 +256,7 @@ func ListResource(r rest.Lister, rw rest.Watcher, scope RequestScope, forceWatch
 		opts := *out.(*api.ListOptions)
```

Comment:
OK, I see now. You're right. However, I'd be more comfortable with this code if it explicitly made an api.ListOptions, and used Convert() instead of ConvertToVersion.